server/grpc_service.go

```diff
@@ -176,6 +183,49 @@ func (s *GrpcServer) Tso(stream pdpb.PD_TsoServer) error {
 		}
 
 		streamCtx := stream.Context()
+		if !s.IsTSOEnabled() {
+			_, forwardedHost, err := s.GetServicePrimaryAddr(ctx, "tso")
+			if err != nil {
+				return errors.WithStack(err)
+			}
+			if forwardStream == nil || lastForwardedHost != forwardedHost {
+				if cancel != nil {
+					cancel()
+				}
+				client, err := s.getDelegateClient(s.ctx, forwardedHost)
+				if err != nil {
+					return err
+				}
+				log.Info("create tso mcs client forward stream", zap.String("forwarded-host", forwardedHost))
+				forwardStream, cancel, err = s.createMCSTSOForwardStream(client)
+				if err != nil {
+					return err
+				}
+				lastForwardedHost = forwardedHost
+				errCh = make(chan error, 1)
+				go forwardMCSTSOClientToServer(forwardStream, server, errCh)
+			}
+			header := request.GetHeader()
+			forwardRequest := &tsopb.TsoRequest{
+				Header: &tsopb.RequestHeader{
+					ClusterId: s.clusterID,
+					SenderId:  header.GetSenderId(),
```

Comment:
shouldn't be zero. I'll ask the serviceless team to provide this value

Reply:
thanks

Reply:
done